### PR TITLE
Delete Strings::format and no longer used unlocalized infrastructure.

### PR DESCRIPTION
**Pull request by BillyONeal**
This PR is the ultimate result of merging the following localization PRs:

* https://github.com/microsoft/vcpkg-tool/pull/893
* https://github.com/microsoft/vcpkg-tool/pull/913
* https://github.com/microsoft/vcpkg-tool/pull/914
* https://github.com/microsoft/vcpkg-tool/pull/919
* https://github.com/microsoft/vcpkg-tool/pull/920
* https://github.com/microsoft/vcpkg-tool/pull/921
* https://github.com/microsoft/vcpkg-tool/pull/922
* https://github.com/microsoft/vcpkg-tool/pull/924
* https://github.com/microsoft/vcpkg-tool/pull/925
* https://github.com/microsoft/vcpkg-tool/pull/928
* https://github.com/microsoft/vcpkg-tool/pull/936
* https://github.com/microsoft/vcpkg-tool/pull/937

This fixes all known localization problems and makes future ones less likely by mostly removing the infrastructure to print unlocalized messages outright.

Audited:

* `print2` (deleted)
* `vcpkg::printf` (deleted)
* `Checks::not-msg` (deleted)
* `Strings::format` (deleted)
* `ExpectedS` (deleted)
* `Strings::append` (retained)
* `Strings::concat` (retained)
* `add_generic_error` (retained)

### CODE REVIEW

File: src/vcpkg/cmakevars.cpp [hunk 1]
```diff
@@ -91,25 +91,26 @@ namespace vcpkg::CMakeVars
         const auto& fs = paths.get_filesystem();
         std::string extraction_file;
```
Comment (ras0219-msft) [inline src/vcpkg/cmakevars.cpp:92]: We could construct extraction_file with the initial prelude.

File: src/vcpkg/commands.dependinfo.cpp [hunk 2]
```diff
@@ -193,19 +193,17 @@ namespace vcpkg::Commands::DependInfo
             for (const auto& package : depend_info)
             {
                 const std::string name = package.package;
-                nodes.append(Strings::format("<Node Id=\"%s\" />", name));
+                fmt::format_to(std::back_inserter(nodes), "<Node Id=\"{}\" />", name);
```
Comment (ras0219-msft) [inline src/vcpkg/commands.dependinfo.cpp:196]: [no change requested] This should really use XmlSerializer...

File: src/vcpkg/export.ifw.cpp [hunk 3]
```diff
@@ -288,23 +288,23 @@ namespace vcpkg::Export::IFW
...
             fs.write_contents(config_xml_file_path,
-                              Strings::format(
+                              fmt::format(
                                   R"###(<?xml version="1.0"?>
 <Installer>
     <Name>vcpkg</Name>
     <Version>1.0.0</Version>
     <StartMenuDir>vcpkg</StartMenuDir>
-    <TargetDir>@RootDir@/src/vcpkg</TargetDir>%s
+    <TargetDir>@RootDir@/src/vcpkg</TargetDir>{%s}
```
Comment (ras0219-msft) [inline src/vcpkg/export.ifw.cpp:307]:
```suggestion
    <TargetDir>@RootDir@/src/vcpkg</TargetDir>{}
```

File: src/vcpkg/export.prefab.cpp [hunk 4]
```diff
@@ -509,9 +509,8 @@ namespace vcpkg::Export::Prefab
             for (const auto& triplet : triplets)
             {
                 const auto listfile =
-                    paths.installed().vcpkg_dir_info() / Strings::format("%s_%s_%s", name, norm_version, triplet) +
-                    ".list";
-                const auto installed_dir = paths.packages() / Strings::format("%s_%s", name, triplet);
+                    paths.installed().vcpkg_dir_info() / fmt::format("{}_{}_{}", name, norm_version, triplet) + ".list";
```
Comment (ras0219-msft) [inline src/vcpkg/export.prefab.cpp:512]:
```suggestion
                    paths.installed().vcpkg_dir_info() / fmt::format("{}_{}_{}.list", name, norm_version, triplet);
```

**Review by ras0219-msft — APPROVED** [review]

File: src/vcpkg/commands.dependinfo.cpp (hunk 2 above)
Comment (BillyONeal) [reply src/vcpkg/commands.dependinfo.cpp:196]: Yeah :(

**Comment by autoantwort** [comment]
This kind of broke the output format of duration. 
Before
```
Detecting compiler hash for triplet arm64-osx...
The following packages will be rebuilt:
    boost-uninstall[core]:arm64-osx -> 1.81.0#2
Removing 1/2 boost-uninstall:arm64-osx
Elapsed time to handle boost-uninstall:arm64-osx: 1.948 ms
Restored 0 package(s) from /Users/leanderSchulten/.cache/vcpkg/archives in 114.9 us. Use --debug to see more details.
Attempting to fetch 1 package(s) from HTTP servers
Restored 0 package(s) from HTTP servers in 206 ms. Use --debug to see more details.
Installing 2/2 boost-uninstall:arm64-osx...
Building boost-uninstall[core]:arm64-osx...
warning: -- Using community triplet arm64-osx. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: /Users/leanderSchulten/git_projekte/vcpkg/triplets/community/arm64-osx.cmake
-- 
Please use the following command when you need to remove all boost ports/components:
    "./vcpkg remove boost-uninstall:arm64-osx --recurse"

-- Performing post-build validation
Uploaded binaries to 1 HTTP remotes.
Stored binary cache: "/Users/leanderSchulten/.cache/vcpkg/archives/66/669209711bf44e71d0470196cd78bc9b486cef0cf363c6cd577f9b70a053e1dc.zip"
Elapsed time to handle boost-uninstall:arm64-osx: 278.4 ms
Total install time: 487.1 ms
```
After:
```
Detecting compiler hash for triplet arm64-osx...
The following packages will be rebuilt:
    boost-uninstall[core]:arm64-osx -> 1.81.0#2
Removing 1/2 boost-uninstall:arm64-osx
Elapsed time to handle boost-uninstall:arm64-osx: 1.88325 ms
Restored 0 package(s) from /Users/leanderSchulten/.cache/vcpkg/archives in 682.708 us. Use --debug to see more details.
Attempting to fetch 1 package(s) from HTTP servers
Restored 0 package(s) from HTTP servers in 210.244292 ms. Use --debug to see more details.
Installing 2/2 boost-uninstall:arm64-osx...
Building boost-uninstall[core]:arm64-osx...
warning: -- Using community triplet arm64-osx. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: /Users/leanderSchulten/git_projekte/vcpkg/triplets/community/arm64-osx.cmake
-- 
Please use the following command when you need to remove all boost ports/components:
    "./vcpkg remove boost-uninstall:arm64-osx --recurse"

-- Performing post-build validation
Uploaded binaries to 1 HTTP remotes.
Stored binary cache: "/Users/leanderSchulten/.cache/vcpkg/archives/ea/ea1a7b9817b0c937939d837215893858a607c709475ce50a9119f085c3cf5089.zip"
Elapsed time to handle boost-uninstall:arm64-osx: 293.621666 ms
Total install time: 506.937584 ms
```

**Comment by autoantwort** [comment]
Since this PR `vcpkg depend-info` crashes with 
```
vcpkg has crashed. Please create an issue at https://github.com/microsoft/vcpkg containing a brief summary of what you were trying to do and the following information.

Version=2999-12-31-unknownhash-debug
EXCEPTION=unmatched '}' in format string
CMD=
vcpkg|
depend-info|
--dot|
vtk|

```

File: src/vcpkg/commands.integrate.cpp [hunk 5]
```diff
@@ -466,7 +465,7 @@ namespace vcpkg::Commands::Integrate
                        .string_arg("-ExecutionPolicy")
                        .string_arg("Bypass")
                        .string_arg("-Command")
-                       .string_arg(Strings::format("& {& '%s' }", script_path));
+                       .string_arg(fmt::format("& {& '{}' }", script_path));
```
Comment (autoantwort) [inline src/vcpkg/commands.integrate.cpp:468]: Probably the reason for the following crash: https://github.com/microsoft/vcpkg/issues/30283

Comment (FrankXie05) [reply src/vcpkg/commands.integrate.cpp:468]: It can solve the problem of `{}` in the parsing path.

**Pull request merged**